packages/anoncreds/src/protocols/proofs/v1/V1ProofProtocol.ts

```diff
@@ -416,6 +416,12 @@ export class V1ProofProtocol extends BaseProofProtocol implements ProofProtocol<
 
     let proofRecord = await this.findByThreadAndConnectionId(agentContext, proofRequestMessage.threadId, connection?.id)
 
+    if (!proofRecord) {
+      // Proof request not bound to any connection: requests_attach in OOB msg
```

Comment:
This doesn't mean the proof record is not bound to any connection, and it means if i know the thread id i respond to this exchange even though not involved. 

We should update this logic to be fully integrated with the oob module:
- we look for the oob record
- if oob is connectionless/now has a connection afterwards we allow the proof record to not have a connection


I'd like to solve this generically for both connectionless/non-connectionless exchanges with attached messages. Is this needed for this PR?